### PR TITLE
Issue13 array highlight code

### PR DESCRIPTION
**Pull request by lilypolonetsky**
As requested by issue 13, I added array highlighting code. 

The `remove` method had to be modified to match the code from the book.

### CODE REVIEW

File: PythonVisualizations/Array.py [hunk 1]
```diff
@@ -224,44 +261,141 @@ def find(self, val):
...
 
+    removeCode = """
+def delete(self, item):
+    for j in range(self.nItems):
+        if self.__a[j] == item:
+            for k in range(j, self.nItems):
+                self.__a[k] = self.__a[k+1]
+            self.nItems -= 1
+            return True
+    return False
```
Comment (JMCanning78) [inline PythonVisualizations/Array.py:298]: This code doesn't quite match what's in the textbook.  The decrement of `nItems` has to happen before the inner for loop, not afterwards.  See Array.py in [02_code](https://drive.google.com/drive/folders/1OXk62CRfU6tsTvxXKUUpIHPUioT5u5U2)

**Comment by JMCanning78** [comment]
Note the difference between the textbook code and what's in this PR is partly due to the way the visualization works.  In the visualization code, there is no `nItems` variable that can be decremented.  The decrement is performed by `removeFromEnd()`.  That's good for showing an animation of the removal of the last item, but is a mismatch to the textbook code.  We can't do that removal animation before the inner for loop that shifts cell values over.  This is an example of what Prof. Waxman was talking about in Slack.  We could try changing the representation of the data in the visualization code to more closely match the textbook code, or just "patch" the visualization to make it look right.
I also suspect that the disappearance of the code from the window is connected to the call to `removeFromEnd()` and the clean up of its call environment...

**Comment by JMCanning78** [comment]
Issue #67 relates to this. To show the change of the `nItems` value, code highlighting is nice, but it should also be visible in the movement of the pointer on the canvas.  Let's try to keep the tasks separate, but eventually the task that's completed last needs to have both the code highlighting and the `nItems` pointer movement coordinated.

**Comment by shirapahmer** [comment]
Miriam and I looked this over to compare the code that is displayed and the code that is in the textbook. (We used the textbook files that Professor Broder had distributed to our data structures class this past semester because we didn't have access to the google drive of the chapters that was uploaded on slack, but we are assuming that they are the same). 
We noticed two things that might be problems:
1. In the highlighted code for find, if it finds the element that is being searched for it returns self.__a[j] because we want to display it in the message at the bottom of the screen. In the textbook however, it just returns j, because that find method is being called by the search() method which returns the actual element and not just the position it was found at. 
2. In the highlighted code for the find button, the method name that is displayed is search() but in the textbook the search() method just calls the find() method. So to match most accurately the highlighted code method should be called find() instead of search().

**Comment by JMCanning78** [comment]
@shirapahmer and @miriamrabinovich, you are absolutely correct in https://github.com/JMCanning78/datastructures-visualization/pull/66#issuecomment-653061302.  I do need to be consistent in the names and code. 
I'll take a shot at correcting those issues, and making commits in the issue13-array-highlight-code branch.

**Comment by JMCanning78** [comment]
OK, I've changed "Find" to "Search", used code highlighting for both `find()` and `get()` methods, animated the removal of the rightmost array cell, fixed several  animation related problems, and shortened several of the status/error messages.  I merged `origin/master` into the issue branch so several other changes have come in.
I'll ask each of you to review, but we may not need all three reviews.

**Comment by JMCanning78** [comment]
Thanks @lilypolonetsky.  @shirapahmer or @miriamrabinovich, please review too and see if the changes addressed your concerns.

**Comment by JMCanning78** [comment]
You're right,  @shirapahmer, the textbook code does prefix the variables with underscores (and discusses the merits of keeping some class attributes private).  I think for the visualizations, however, we can drop the initial underscores.

**Pull request merged**